apps/stats/management/commands/update_counts_from_file.py

```diff
@@ -84,6 +89,27 @@ def handle(self, *args, **options):
         guids_to_addon = (dict(Addon.objects.exclude(guid__isnull=True)
                                             .exclude(type=amo.ADDON_PERSONA)
                                             .values_list('guid', 'id')))
+        # Perf: preload all the versions for each addon, used to validate the
+        # version numbers for a given addon.
+        # This builds a dict where each key (the addon id) has a list of all
+        # its versions as a value.
+        all_versions = Version.objects.values_list('addon_id', 'version')
+        self.addons_versions = {}
+        for addon_id, version in all_versions:
+            self.addons_versions.setdefault(addon_id, [])
+            self.addons_versions[addon_id].append(version)
+
+        # This gives a list of (application IDs, version).
+        appversions = AppVersion.objects.values_list('application', 'version')
+        # We want the application GUID, not the application ID.
+        appversions = [(amo.APPS_ALL[app_id].guid, version)
+                       for app_id, version in appversions]
+        # This builds a dict where each key (the application guid) has a list
+        # of all its versions as a value.
+        self.valid_appversions = {}
+        for app_guid, app_version in appversions:
+            self.valid_appversions.setdefault(app_guid, [])
+            self.valid_appversions[app_guid].append(app_version)
```

Comment:
This will create a dict of less than 42k items (number of addons that are not themes), each of those having as value a list of less than 800 elements (the [noscript addon](https://addons.mozilla.org/fr/firefox/addon/noscript/versions/) has the most versions: 860). The total number of versions is less than 600k.

If an average version field is around 10 chars, this means 6millions chars, which will eat up 6 megabytes or ram if I count correctly. Mysql says that the average field size is 2, and the max is 29.